File: app/src/main/java/com/retech/aspectj_demo/MainActivity.java
package com.retech.aspectj_demo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.retech.mylibrary.NeedPermissions;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getName(new PermissionResult() {
            @Override
            public void onDenied() {

            }

            @Override
            public void onGranted() {

            }
        });
        getPermission();
    }

    @MyAnnotation(getArgs = {"hhhhh", "rrrrrrrrr"})
    public void getName(PermissionResult permissionResult) {
        Log.e("glz", "my name is chen");
    }

    @NeedPermissions("ssd")
    public void getPermission() {
        Log.e("glz", "permissionssssss");
    }

    public void getAge(String... name) {
        int a = 10;
        int b = 11;
        int c = a > b ? a : b;
    }
}

File: myktapplication/src/main/java/com/retech/myktapplication/MainActivity.kt
package com.retech.myktapplication

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        getName()
    }

    @NeedPermission("sfs")
    fun getName() {
        val a = 10
        val b = 11
        val c = if (a > b) a else b
    }

}
